scanpointgenerator/generators/zipgenerator.py

```diff
@@ -0,0 +1,44 @@
+from annotypes import Anno, deserialize_object, Array, Sequence, Union
+from scanpointgenerator.core import Generator
+
+with Anno("List of Generators to zip"):
+    AGenerators = Array[Generator]
+UGenerators = Union[AGenerators, Sequence[Generator], Generator]
+
+
+@Generator.register_subclass(
+    "scanpointgenerator:generator/ZipGenerator:1.0")
+class ZipGenerator(Generator):
+    """ Zip generators together, combining all generators into one """
+
+    def __init__(self, generators):
+        # type: (UGenerators) -> None
+        self.generators = AGenerators([deserialize_object(g, Generator)
+                                      for g in generators])
+
+        units = []
+        axes = []
+        size = self.generators[0].size
+
+        for generator in self.generators:
+            assert generator.axes not in axes, "You cannot zip generators " \
+                                               "on the same axes"
```

Comment:
I think, in general, I have not aligned when splitting over multiple lines in this code, but just used a double-indent.

Reply:
I've aligned here to stop pycharm complaining about a PEP-8: continuation line ocer-indented for visual indent